collect_app/src/main/java/org/odk/collect/android/preferences/stores/AdminPreferencesDataStore.kt

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.preferences.stores
+
+import androidx.preference.PreferenceDataStore
+import org.odk.collect.android.preferences.PreferencesDataSourceProvider
+
+class AdminPreferencesDataStore(val preferencesDataSourceProvider: PreferencesDataSourceProvider) : PreferenceDataStore() {
```

Comment:
It feels we could just have a `PreferencesDataSourceDataStore` that takes a `PreferenceDataSource` in its constructor rather than specific implementations for admin and general. Is there a blocker for doing that?

Reply:
I'm also thinking we might want to rename our `PreferenceDataSource` interface to avoid any confusion. I guess `PreferenceDataStore` will only be used in the context of `PreferenceFragment` though.

Reply:
Yeah I was thinking about this but somehow ended up with two implementation I can't recall the reason but now it's seems doable, what I did and now I have one PreferenceDataStore.